packages/marshal/src/helpers/copyRecord.js

```diff
@@ -24,25 +24,32 @@ export const CopyRecordHelper = harden({
   styleName: 'copyRecord',
 
   canBeValid: (candidate, check) => {
-    const reject = details => check(false, details);
     const proto = getPrototypeOf(candidate);
     if (proto !== objectPrototype && proto !== null) {
-      return reject(X`Unexpected prototype for: ${candidate}`);
+      return !!check && check(false, X`Unexpected prototype for: ${candidate}`);
     }
     const descs = getOwnPropertyDescriptors(candidate);
     const descKeys = ownKeys(descs);
 
     for (const descKey of descKeys) {
       if (typeof descKey !== 'string') {
         // Pass by copy
-        return reject(
-          X`Records can only have string-named own properties: ${candidate}`,
+        return (
+          (!!check &&
+          check(
+            false,
+            X`Records can only have string-named own properties: ${candidate}`,
+          ))
```

Comment:
This pattern would arguably be better with ES2020 optional chaining:
```suggestion
          !!check?.(
            false,
            X`Records can only have string-named own properties: ${candidate}`,
          )
```